app/services/documentation_generator.rb

```diff
@@ -0,0 +1,371 @@
+                                  description: "The field that the error message would be added to.",
+                                },
+                              },
+                              required: %w[error_message field],
+                            },
+                          },
+                          from: {
+                            type: :number,
+                            description: "the year from which the validation starts. If this validation runs for logs with a startdate after a certain year, specify that year here, only if it is not specified in the validation method, leave this field blank",
+                          },
+                          to: {
+                            type: :number,
+                            description: "the year in which the validation ends. If this validation runs for logs with a startdate before a certain year, specify that year here, only if it is not specified in the validation method, leave this field blank",
+                          },
+                          validation_type: {
+                            type: :string,
+                            enum: %w[presence format minimum maximum range inclusion length other],
+                            description: "The type of validation that is being performed. This should be one of the following: presence (validates that the question is answered), format (validates that the answer format is valid), minimum (validates that entered value is more than minimum allowed value), maximum (validates that entered value is less than maximum allowed value), range (values must be between two values), inclusion (validates that the values that are not allowed arent selected), length (validates the length of the answer), other",
+                          },
+                          other_validated_models: {
+                            type: :string,
+                            description: "Comma separated list of any other models (other than log) that were used in this validation. These are possible models (only add a value to this field if other validation models are one of these models): User, Organisation, Scheme, Location, Organisation_relationship, LaRentRange. Only leave this blank if no other models were used in this validation.",
+                          },
+                        },
+                        required: %w[case_description errors validation_type other_validated_models],
+                      },
+                    },
+                  },
+                  required: %w[description cases],
+                },
+              },
+            },
+          ],
+          tool_choice: { type: "function", function: { name: "create_documentation_for_given_validation" } },
+        },
+      )
+    rescue StandardError => e
+      Rails.logger.error("Failed to describe #{meth} for #{form.start_date.year}")
+      Rails.logger.error("Error #{e.message}")
+      sleep(15)
```

Comment:
What's the sleep for?

Reply:
Oh, yeah, there's a limit on the tokens we can use per minute, for 3.5 it's kinda low and I'm expecting to run these tasks a few times anyway to populate everything, but the sleep kinda helps with that